craft_parts/parts.py

```diff
@@ -16,15 +16,95 @@
 
 """Definitions and helpers to handle parts."""
 
-import copy
 from pathlib import Path
 from typing import Any, Dict, List, Optional, Sequence, Set
 
+from pydantic import BaseModel, Field, ValidationError
+
 from craft_parts import errors
 from craft_parts.dirs import ProjectDirs
 from craft_parts.steps import Step
 
 
+class PartSpec(BaseModel):
+    """The part specification data."""
+
+    plugin: Optional[str]
```

Comment:
I'd suggest being explicit about the default for the optionals.
```suggestion
    plugin: Optional[str] = None
```

Is plugin really optional?

Reply:
Now it is. If plugin is not specified, we use the part name instead. This is used e.g. in the charmcraft `bundle` part.